src/Blocks/BlockEnchantmentTable.h

```diff
@@ -34,6 +34,11 @@ class cBlockEnchantmentTableHandler :
 		UNUSED(a_Meta);
 		return 29;
 	}
+
+	virtual double GetBlockHeight(void) override
+	{
+		return 12.0 / 16.0;
```

Comment:
I don't think we need to worry about compilers that don't optimize this away, because such compilers won't optimize anything at all and the performance would be terrible anyway. All modern compilers (and we use C++11, so we need a modern compiler) do the optimization.